pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -37,131 +39,179 @@ func newTestManager() (*basicManager, *podtest.FakeMirrorClient) {
-		},
-	}
-
-	expectedPods := []*v1.Pod{
-		{
+		}
+		staticPod = &v1.Pod{
 			ObjectMeta: metav1.ObjectMeta{
-				UID:         "999999999",
-				Name:        "taco",
-				Namespace:   "default",
-				Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "api"},
+				UID:         types.UID("static-pod-uid"),
+				Name:        "mirror-static-pod-name",
+				Namespace:   metav1.NamespaceDefault,
+				Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "file"},
 			},
-		},
-		staticPod,
-	}
-	updates := append(expectedPods, mirrorPod)
-	podManager, _ := newTestManager()
-	podManager.SetPods(updates)
-
-	// Tests that all regular pods are recorded correctly.
-	actualPods := podManager.GetPods()
-	if len(actualPods) != len(expectedPods) {
-		t.Errorf("expected %d pods, got %d pods; expected pods %#v, got pods %#v", len(expectedPods), len(actualPods),
-			expectedPods, actualPods)
-	}
-	for _, expected := range expectedPods {
-		found := false
-		for _, actual := range actualPods {
-			if actual.UID == expected.UID {
-				if !reflect.DeepEqual(&expected, &actual) {
-					t.Errorf("pod was recorded incorrectly. expect: %#v, got: %#v", expected, actual)
-				}
-				found = true
-				break
-			}
 		}
-		if !found {
-			t.Errorf("pod %q was not found in %#v", expected.UID, actualPods)
+		normalPod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       types.UID("normal-pod-uid"),
+				Name:      "normal-pod-name",
+				Namespace: metav1.NamespaceDefault,
+			},
 		}
+	)
+
+	testCase := []struct {
+		name         string
+		podList      []*v1.Pod
+		wantPod      *v1.Pod
+		expectPods   []*v1.Pod
+		expectGetPod *v1.Pod
+		expectUID    types.UID
+	}{
+		{
+			name:         "Get normal pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      normalPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: normalPod,
+			expectUID:    types.UID("static-pod-uid"),
+		},
+		{
+			name:         "Get static pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      staticPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: staticPod,
+			expectUID:    types.UID("static-pod-uid"),
+		},
 	}
-	// Tests UID translation works as expected. Convert static pod UID for comparison only.
-	if uid := podManager.TranslatePodUID(mirrorPod.UID); uid != kubetypes.ResolvedPodUID(staticPod.UID) {
-		t.Errorf("unable to translate UID %q to the static POD's UID %q; %#v",
-			mirrorPod.UID, staticPod.UID, podManager.mirrorPodByUID)
-	}
+	for _, test := range testCase {
+		t.Run(test.name, func(t *testing.T) {
+			podManager, _ := newTestManager()
+			podManager.SetPods(test.podList)
+			actualPods := podManager.GetPods()
+			sort.Slice(actualPods, func(i, j int) bool {
+				return actualPods[i].Name < actualPods[j].Name
+			})
+			if !reflect.DeepEqual(actualPods, test.expectPods) {
```

Comment:
Use `cmp.Diff` instead of `reflect.DeepEqual` 
ref: https://github.com/kubernetes/community/blob/d8d3aec2502be3e90b81c91e42fba65539ae6376/sig-scheduling/CONTRIBUTING.md?plain=1#L107

Reply:
thanks for reminding this!